runtime/lua/vim/lsp/inlay_hint.lua

```diff
@@ -260,15 +393,19 @@ api.nvim_set_decoration_provider(namespace, {
   end,
 })
 
-return setmetatable(M, {
-  __call = function(_, bufnr, enable_)
-    vim.validate({ enable = { enable_, { 'boolean', 'nil' } }, bufnr = { bufnr, 'number' } })
-    if enable_ then
-      enable(bufnr)
-    elseif enable_ == false then
-      disable(bufnr)
-    else
-      toggle(bufnr)
-    end
-  end,
-})
+--- Enable/disable/toggle inlay hints for a buffer
+---
+--- @param bufnr (integer) Buffer handle, or 0 for current
+--- @param enable (boolean|nil) true/false to enable/disable, nil to toggle
+function M.toggle(bufnr, enable)
```

Comment:
"toggle" is not a verb in our existing apis, but "enable" is well established. Can avoid adding a new verb, helps discoverability.

```suggestion
function M.enable(bufnr, enable)
```

Reply:
Those cases are where `.enable()` and `.disable()` are separately exported, which seems less efficient.

I'd probably not roll it into `.enable(bool)` since that might be more confusing (uncanny valley) than `.toggle(bool)`

I'm not opposed to `.enable()` + `disable()` + `bool enabled()`, although it's not a very widespread precedent (only see in `diagnostic.lua` and `loader.lua`), maybe worth considering this instance on its own merit (two funcs or one func that does more?)?

Reply:
It does seem like in other domains (e.g. REST) people prefer two endpoints over one toggle endpoint, so that's a point in favor of `.enable()` + `.disable()`. I'll do it

Reply:
Agreed. Although an unprecedented verb isn't ideal, using a function called `enable` to disable feels even less intuitive.

> I'm not opposed to `.enable()` + `disable()` + `bool enabled()`

Idk I don't like the idea of having a bunch of redundant functions...

Reply:
> Those cases are where .enable() and .disable() are separately exported, which seems less efficient.

Those are just accidents. When don't need a new verb. And we can establish `enable()` as the omni-interface for (1) enable, (2) disable, (3) toggle. Having 2 or 3 separate functions for this trivial use-case is silly.

Reply:
For what's worth my vote would also go for `toggle`  instead of a `enable` that can also disable. 

It's definition is a pretty good much, and its not like it isn't used in the code base (granted not as API as far as I can tell, but we were just talking about a win_toggle the other day as well)

```
↪  rg toggle | wc -l
214
```

Reply:
> but we were just talking about a win_toggle the other day as well

The name in that case was more of a placeholder.

The most important thing is that we have 1 function instead of 3 functions for everything that needs to be "toggled". So I can live with "toggle", can we agree to deprecate all the `enable`/`disable` cases in the API?

Reply:
Thinking on this more,  I really don't think we should go with toggle(). Instead:

    enable(..., enable: boolean | 'toggle')

"Toggle" is a minor convenience, it's not the common case that people are looking for. "Enable" is extremely common', naming the interface `toggle` seems quirky and uncommon.

To support the convenience of toggling, we can support a 'toggle' string argument, and use boolean for enable/disable.

Reply:
Didn't we have that exact same discussion already for inlay_hints? And didn't we resolve that? So can we not just follow that precedent and save another discussion?

Reply:
`enable(..., is_enabled())` makes sense as a toggle because we need `is_enabled` anyway. So if that was the result of the previous discussion then yes, we can avoid a new discussion.

Reply:
Previous discussion is at: https://github.com/neovim/neovim/pull/23984

Looks like we went with the toggle there